### PR TITLE
foreman plugins to JJB and GHPRB

### PR DESCRIPTION
**Pull request by komidore64**
I am splitting out all of the foreman plugin pull-request tests into their own individual jobs to facilitate using JJB and GHPRB.
Gone are the days of `test_plugin_pull_request` and in with the new Jenkins Job Builder `job-template` hotness.

[test_plugin_pull_request](http://ci.theforeman.org/job/test_plugin_pull_request/) will need removing after this is merged.

These new jobs follow the naming scheme of `<plugin_name>-pull-request`.

Each job is wired for GHPRB, so a bunch of Github webhooks will need to be added for pull-requests to start properly firing off jobs.

The following plugins will need a new webhook added:
- [x] [foreman_ansible](https://github.com/theforeman/foreman_ansible)
- [x] [foreman_bootdisk](https://github.com/theforeman/foreman_bootdisk)
- [x] [foreman_digitalocean](https://github.com/theforeman/foreman-digitalocean)
- [x] [foreman_discovery](https://github.com/theforeman/foreman_discovery)
- [x] [foreman_docker](https://github.com/theforeman/foreman_docker)
- [x] [foreman_expire_hosts](https://github.com/theforeman/foreman_expire_hosts)
- [x] [foreman_host_extra_validator](https://github.com/theforeman/foreman_host_extra_validator)
- [x] [foreman_monitoring](https://github.com/theforeman/foreman_monitoring)
- [x] [foreman_omaha](https://github.com/theforeman/foreman_omaha)
- [x] [foreman_openscap](https://github.com/theforeman/foreman_openscap)
- [x] [foreman_pipeline](https://github.com/theforeman/foreman_pipeline)
- [x] [foreman_remote_execution](https://github.com/theforeman/foreman_remote_execution)
- [x] [foreman_salt](https://github.com/theforeman/foreman_salt)
- [x] [foreman_setup](https://github.com/theforeman/foreman_setup)
- [x] [foreman-tasks](https://github.com/theforeman/foreman-tasks)
- [x] [foreman_templates](https://github.com/theforeman/foreman_templates)
- [x] [foreman_userdata](https://github.com/theforeman/foreman_userdata)
- [x] [puppetdb_foreman](https://github.com/theforeman/puppetdb_foreman)

If you want to re-run the job again for your initially submitted a pull-request, please comment `[test <plugin_name>]`. make sure you have the proper **rubygem name** of the plugin because that is its technical name. `foreman-tasks` (hyphen instead of underscore) and `foreman_digitalocean` (hypen in the github repo name, but underscore in the rubygem name) are the odd ones out.

**EDIT**: Also before merging this, I need to create a PR removing all of the above plugins from the pr_scanner on https://github.com/theforeman/prprocessor

### CODE REVIEW

**Comment by komidore64** [comment]
annnnnnnnd i forgot to remove the plugins from the PR second in the node's puppet. i'll do that in a bit.

File: puppet/modules/slave/manifests/init.pp [hunk 1]
```diff
@@ -50,7 +50,6 @@
     group  => 'jenkins',
   }
   slave::pr_test_config { [
-    'foreman_ansible',
     'foreman_bootdisk',
```
Comment (ekohl) [inline puppet/modules/slave/manifests/init.pp]: I think this and all other plugins you've converted should be moved.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/builders/foreman-plugins-pull-request.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+- builder:
+    name: plugin-pull-request
+    builders:
+      - shell: |+
+          #!/bin/bash -ex
+
+          # check out the pull request
+          [ -e plugin ] && rm -rf plugin/
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/builders/foreman-plugins-pull-request.yaml]: `rm -rf plugin/` will also exit with 0 if it doesn't exist.

File: puppet/modules/slave/manifests/init.pp (hunk 1 above)
Comment (komidore64) [reply puppet/modules/slave/manifests/init.pp]: yeah as soon as i threw up the PR, i had just remembered those.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/builders/foreman-plugins-pull-request.yaml (hunk 2 above)
Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/builders/foreman-plugins-pull-request.yaml]: oh nice! that's much simpler

File: puppet/modules/slave/manifests/init.pp (hunk 1 above)
Comment (komidore64) [reply puppet/modules/slave/manifests/init.pp]: updated

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/builders/foreman-plugins-pull-request.yaml (hunk 2 above)
Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/builders/foreman-plugins-pull-request.yaml]: updated

**Comment by komidore64** [comment]
plz2merge so that @ares can get his foreman_discovery PRs tested

**Comment by komidore64** [comment]
oh cool! thanks for doing that prprocessor pull-request @ekohl.

**Pull request merged**